File: flutter/lib/domain/post/usecases/createPostUseCase.dart
import '../postRepostiory.dart';

import '../post.dart';

class CreatePostUseCase {
  PostRepository postRepository;

  CreatePostUseCase(PostRepository postRepository) {
    this.postRepository = postRepository;
  }

  Future<Post> run(Post post) {
    return postRepository.createPost(post);
  }
}

File: web/src/domain/user/usecases/CreatePostUseCase.ts
import Post from "../../post/Post";
import PostRepository from "../../post/PostRepository";

export default class CreatePostUseCase {
    postRepository: PostRepository;

    constructor(postRepository: PostRepository) {
        this.postRepository = postRepository;
    }

    async run(post: Post): Promise<Post> {
        return await this.postRepository.createPost(post);
    }
}